operator/src/main/java/dev/responsive/k8s/controller/ControllerProtoFactories.java

```diff
@@ -63,7 +68,13 @@ private static ApplicationPolicy policyFromK8sResource(final ResponsivePolicySpe
     return builder.build();
   }
 
-  private static String getFullApplicationName(final ResponsivePolicy policy) {
-    return policy.getSpec().getApplicationNamespace() + "/" + policy.getSpec().getApplicationName();
+  private static String getFullApplicationName(
+      final String environment,
+      final ResponsivePolicy policy
+  ) {
+    final String prefix = environment.isEmpty() ? "" : environment + "/";
```

Comment:
thoughts on making this required?

Reply:
I thought about it. I don't feel super strongly either way. I erred on the side of not breaking stuff that's already running.